### PR TITLE
Adds a way for panes to filter compilers

### LINKED ISSUE #2730: [BUG] "Execution only" pane can select targets that don't support execution
**Describe the bug**
User can be facing a configuration that is hard to understand ("What is wrong here???")

**To Reproduce**
Enter simple code
- Add >> Execution only (should show x86 exec result)
- change the compiler to something NOT supporting exec (eg. any KVX)
- "Could not execute the program"

**Expected behavior**
The compiler selection diag should simply not show compilers where there's no exec support

**Link**
https://godbolt.org/z/zaYaPKnMY

### PR DESCRIPTION
**Pull request by RubenRBS**
I don't really like this solution, but I haven't been able to come
 up with a cleaner one, so unless someone finds a better way,
 this will have to do

Closes #2730

### CODE REVIEW

File: static/panes/executor.js [hunk 1]
```diff
@@ -107,6 +108,10 @@ function Executor(hub, container, state) {
     });
 }
 
+Executor.prototype.compilerIsVisible = function (compiler) {
+    return compiler.supportsExecute;
```
Comment (partouf) [inline static/panes/executor.js:112]: should also be there if the current compiler is the one being displayed

**Comment by partouf** [comment]
I think this is a good solution

**Comment by mattgodbolt** [comment]
I think this is a great solution! +1 to @partouf 's comment but LGTM!

**Review by mattgodbolt — APPROVED** [review]
+1 to @partouf's point. I think that can be achieved in line 142 by putting the compilerIsVisible check inside the parens

File: static/compiler-picker.js [hunk 2]
```diff
@@ -133,9 +138,9 @@ CompilerPicker.prototype.initialize = function (langId, compilerId) {
 CompilerPicker.prototype.getOptions = function (langId, compilerId) {
     var favorites = this.getFavorites();
     return _.chain(this.compilerService.getCompilersForLang(langId))
-        .filter(function (e) {
-            return !e.hidden || e.id === compilerId;
-        })
+        .filter(_.bind(function (e) {
+            return this.compilerIsVisible(e) && (!e.hidden || e.id === compilerId);
```
Comment (mattgodbolt) [inline static/compiler-picker.js:142]: (specifically:

return (this.compilerIsVisible && ! e.hidden) || e.id === compilerId;
)

**Comment by RubenRBS** [comment]
Great point. Thanks! :)

**Pull request merged**